backend/app/services/music_service.py

```
@@ -2,66 +2,78 @@
+    }
 
-    async def generate_music(self, prompt: str, lantern_id: str):
-        user = await self.user_repo.find_by_lantern_id(lantern_id)
-        if not user:
-            raise NotFoundError(f"Lantern ID {lantern_id} not found.")
+    try:
+        resp = httpx.post(url, json=payload, timeout=1000.0)
+        resp.raise_for_status()
+        body = resp.json()
+    except Exception as e:
+        raise AIResponseProcessingError(f"Failed to fetch AI response: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)